plugin/trino-kafka/src/test/java/io/trino/plugin/kafka/TestKafkaWithConfluentLatestConnectorSmokeTest.java

```diff
@@ -0,0 +1,1299 @@
+import static io.trino.spi.type.IntegerType.INTEGER;
+import static io.trino.spi.type.RealType.REAL;
+import static io.trino.spi.type.TimeType.TIME_MICROS;
+import static io.trino.spi.type.TimeType.TIME_MILLIS;
+import static io.trino.spi.type.TimestampType.TIMESTAMP_MICROS;
+import static io.trino.spi.type.TimestampType.TIMESTAMP_MILLIS;
+import static io.trino.spi.type.VarbinaryType.VARBINARY;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static io.trino.type.DateTimes.PICOSECONDS_PER_MICROSECOND;
+import static io.trino.type.DateTimes.PICOSECONDS_PER_MILLISECOND;
+import static java.lang.String.format;
+import static java.util.Locale.ENGLISH;
+import static org.apache.kafka.clients.producer.ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG;
+import static org.apache.kafka.clients.producer.ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.testng.Assert.assertTrue;
+
+public class TestKafkaWithConfluentLatestConnectorSmokeTest
+        extends BaseConnectorSmokeTest
+{
+    @ManageTestResources.Suppress(because = "This class is stateless, it just invokes one method.")
+    private static final LongSerializer LONG_SERIALIZER = new LongSerializer();
+    @ManageTestResources.Suppress(because = "This class is stateless, it just invokes one method.")
+    private static final IntegerSerializer INTEGER_SERIALIZER = new IntegerSerializer();
+
+    private TestingKafka testingKafka;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        testingKafka = closeAfterClass(TestingKafka.createWithSchemaRegistry());
+        return KafkaWithConfluentSchemaRegistryQueryRunner.builder(testingKafka)
+                .setTables(REQUIRED_TPCH_TABLES)
+                .setExtraKafkaProperties(ImmutableMap.<String, String>builder()
+                        .put("kafka.confluent-subjects-cache-refresh-interval", "1ms")
+                        .buildOrThrow())
+                .build();
+    }
+
+    @SuppressWarnings("DuplicateBranchesInSwitch")
+    @Override
+    protected boolean hasBehavior(TestingConnectorBehavior connectorBehavior)
+    {
+        switch (connectorBehavior) {
+            case SUPPORTS_CREATE_SCHEMA:
+                return false;
+
+            case SUPPORTS_CREATE_TABLE:
+            case SUPPORTS_RENAME_TABLE:
+            case SUPPORTS_CREATE_MATERIALIZED_VIEW:
+            case SUPPORTS_MERGE:
+            case SUPPORTS_UPDATE:
+            case SUPPORTS_CREATE_VIEW:
+                return false;
+
+            default:
+                return super.hasBehavior(connectorBehavior);
+        }
+    }
+
+    @Override
+    public void testInsert()
+    {
+        assertThatThrownBy(super::testInsert)
+                .hasMessage("Cannot test INSERT without CREATE TABLE, the test needs to be implemented in a connector-specific way");
+        // TODO implement the test for Kafka
```

Comment:
reminder

Reply:
Custom tests for insert are in `TestKafkaWithConfluentLatestConnectorSmokeTest`.